flink-tests/src/test/java/org/apache/flink/test/checkpointing/StatefulJobSnapshotMigrationITCase.java

```diff
@@ -136,6 +136,9 @@ public void testSavepoint() throws Exception {
         switch (snapshotSpec.getStateBackendType()) {
             case StateBackendLoader.ROCKSDB_STATE_BACKEND_NAME:
                 env.setStateBackend(new EmbeddedRocksDBStateBackend());
+                // disable changelog backend for now to ensure determinism in test data generation
+                // (see FLINK-31766)
+                env.enableChangelogStateBackend(false);
```

Comment:
nit: this change duplicates line 155, one can be removed

nit: can be scoped to `executionMode == ExecutionMode.CREATE_SNAPSHOT` only

Reply:
good catch. I'm gonna update it.